File: app/Shared/HopConstants.swift
import Foundation

enum HopConstants {
    static let appDisplayName = "ɹǝddoH"

    static var appVersion: String {
        Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? "2.5.2"
    }

    static let appGroupID = "group.com.aengix.hopper"
    static let mainBundleID = "com.aengix.hopper"
    static let tunnelBundleID = "com.aengix.hopper.tunnel"

    static let profileStoreFileName = "hopper-profiles.json"
    static let tunnelLastErrorFileName = "last-tunnel-error.txt"
    static let deviceIDKey = "hopper-device-id"

    static let tunnelIPv4Mask = "255.255.255.0"
    static let tunnelRemoteAddress = "10.64.0.1"

    static let tunnelMTU = 1280

    static let defaultSSHPort = 22
    static let defaultInstallDir = "~/hopper"

    static let serverInstallRepo = "ZonD80/hopper"
    static let serverInstallRef = "main"
    static var serverInstallURL: String {
        "https://raw.githubusercontent.com/\(serverInstallRepo)/\(serverInstallRef)/server/install.sh"
    }

    /// First overlay address octet for hop index 0 (entry).
    static let overlayNodeOctet = 10
}

File: app-android/app/src/main/java/com/aengix/hopper/model/HopConstants.kt
package com.aengix.hopper.model

import android.content.Context

object HopConstants {
    const val APP_DISPLAY_NAME = "ɹǝddoH"
    const val PROFILE_STORE_FILE_NAME = "hopper-profiles.json"
    const val TUNNEL_LAST_ERROR_FILE_NAME = "last-tunnel-error.txt"
    const val DEVICE_ID_KEY = "hopper-device-id"

    const val TUNNEL_IPV4_MASK_BITS = 24
    const val TUNNEL_REMOTE_ADDRESS = "10.64.0.1"
    const val TUNNEL_IPV6_SINKHOLE = "fd00::1"

    const val TUNNEL_MTU = 1280

    const val DEFAULT_SSH_PORT = 22
    const val DEFAULT_INSTALL_DIR = "~/hopper"

    const val SERVER_INSTALL_REPO = "ZonD80/hopper"
    const val SERVER_INSTALL_REF = "main"
    const val SERVER_INSTALL_URL =
        "https://raw.githubusercontent.com/$SERVER_INSTALL_REPO/$SERVER_INSTALL_REF/server/install.sh"

    const val OVERLAY_NODE_OCTET = 10

    fun appVersion(context: Context): String = runCatching {
        @Suppress("DEPRECATION")
        context.packageManager.getPackageInfo(context.packageName, 0).versionName
    }.getOrNull() ?: HopVersion.manifest.version
}
